### PR TITLE
{python3_24} Python3: fix logger

### PR DESCRIPTION
**Pull request by cjmayo**

### CODE REVIEW

File: linkcheck/logger/__init__.py [hunk 1]
```diff
@@ -323,7 +324,7 @@ def writeln (self, s=u"", **args):
         """
         Write string to output descriptor plus a newline.
         """
-        self.write(u"%s%s" % (s, unicode(os.linesep)), **args)
+        self.write(u"%s%s" % (s, str_text(os.linesep)), **args)
```
Comment (mgedmin) [inline linkcheck/logger/__init__.py]: TBH `u"%s%s" % (s, os.linesep)` should work just as well.

Comment (anarcat) [reply linkcheck/logger/__init__.py]: but not in py2?

Comment (mgedmin) [reply linkcheck/logger/__init__.py]: no, it should work in all versions of Python.

On Python 2

- `u"...%s..." % str_containing_pure_ascii` is safe and returns Unicode.

- `u"...%s..." % str_containing_arbitrary_user_supplied_value` is dangerous and might raise a UnicodeDecodeError.

- (and `u"...{}...".format(str_containing_pure_ascii)` would always raise an exception.)

Comment (anarcat) [reply linkcheck/logger/__init__.py]: okay, then let's just do that instead here, no?

Comment (mgedmin) [reply linkcheck/logger/__init__.py]: Eh, it's a very minor thing and I already regret bringing it up.

Comment (mgedmin) [reply linkcheck/logger/__init__.py]: (It's important to not use `unicode(os.linesep)` because `unicode` is not a defined name on Python 3.)

**Pull request merged**